### ISSUE TITLE
DatagridHeaderRow: Received `false` for a non-boolean attribute `role` warning

### ISSUE THREAD

**Issue by flannanl** [carbon-design-system/ibm-products#4310]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.
When rendering `Datagrid`, a React warning, `Received "false" for a non-boolean attribute "role" warning` is logged in the console. Inspecting the source code, it seems it is caused by this line in the `DatagridHeaderRow` component. Note that this seems to be only happening in `v1`.

https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/ibm-products/src/components/Datagrid/Datagrid/DatagridHeaderRow.js
```
{...header.getHeaderProps({ role: false })}
```

> Is this issue related to a specific component?
Datagrid

> What did you expect to happen? What happened instead? What would you like to
> see changed?
No React warning should be logged in console.

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?
v1.70

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
IBM Cloud Projects

## Steps to reproduce the issue

- Render a Datagrid with table headers.
